poppy/wfe.py

```diff
@@ -353,6 +353,151 @@ def get_opd(self, wave):
         phase_screen = np.fft.ifftshift(np.fft.ifft2(np.fft.ifftshift(scaled))).real   # FT of scaled random PSD makes phase screen
 
         phase_screen -= np.mean(phase_screen)  # force zero-mean
-        opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
+        self.opd = phase_screen / np.std(phase_screen) * self.wfe.to(u.m).value  # normalize to wanted input rms wfe
 
-        return opd
+        return self.opd
+
+
+class PowerSpectrumWFE(WavefrontError):
+    """
+    Power spectrum PSD WFE class from characterizing and modeling 
+    optical surface power spectrum and applying optical noise.
```

Comment:
I've added more content about the von Karman PSD model in the comment block. However, I'm not quite sure how to set up the LaTeX for the comments. Joseph pointed me to something I can use, but I'm unsure if only because I don't know how to check my (like with all LaTeX writing practice). If you can let me know how I can check this, I can fix it up.